fastlane/Fastfile

```diff
@@ -134,6 +135,19 @@ before_all do
   # but it has such a negligible overhead that it's worth running it here to
   # keep the individual lanes cleaner.
   configure_apply
+
+  unless is_ci || File.file?(USER_ENV_FILE_PATH)
+    example_path = 'fastlane/user.env-example'
+    UI.user_error! <<~MSG
+      #{ENV_FILE_NAME} not found!
+
+      Please copy #{example_path} to #{USER_ENV_FILE_PATH} and fill in the values for the automation you require.
+
+      cd #{example_path} #{USER_ENV_FILE_PATH}
```

Comment:
```suggestion
      cp #{example_path} #{USER_ENV_FILE_PATH}
```

Reply:
I'm wondering if warning about this systematically (i.e. in the `before_all`, run for every lane even the ones that might not need any of those env vars) is the right approach after all (I know that's what we do for other apps, but maybe this PR is a good occasion to start introducing a better approach?).

What do you think about making that error only printed when a `get_required_env` fails (and `is_ci` is `false`), so that it's only printed _if_ a required env var is actually missing, but not printed nor required if the lane being called ends up not needing any secret env value at all?

---

An alternative approach that I've started to use in Tumblr is that in addition to `get_required_env` helper, we [also have a `require_env_vars ` helper](https://github.tumblr.net/Tumblr/ios/blob/develop/fastlane/lib/helpers.rb#L18-L20) which can take multiple env var names, and whose role is only to check, [at the very start of each lane](https://github.tumblr.net/Tumblr/ios/blob/da7f978a78a8f755eb1cbaa151cfb2f1072e9cde/fastlane/Fastfile#L95), if all those required env vars that the lane will be needed are defined.

The nice aspect of that approach is that it's immediately clear when you look at each lane in the `Fastfile` which env var this lane will be required to be defined, instead of skimming the code of the lane for all the `get_required_env` calls that it contains… which would not be enough anyway, because that lane might require that env var _directly_ but might call helper functions or other lanes which _do_ require additional env vars, or might be calling actions without explicitly passing some env vars (e.g. not explicitly passing the `token: get_required_env('GITHUB_TOKEN')` parameter to an action but instead relying on the fact that `GITHUB_TOKEN` is already the `env_name` of the corresponding `ConfigItem` in that action and that it will use it automatically).

If you go with that approach of adding `require_env_vars` at the top of your lanes to make it more clear what's required for each, then it would then make sense to implement that test (and printing the instructions for the user to create the `USER_ENV_FILE_PATH` file) in the implementation of `require_env_vars` itself (and only if at least one of the env var was not defined), instead of in `before_all`?